.NET/Microsoft.Recognizers.Text.DateTime/Extractors/BaseTimeZoneExtractor.cs

```diff
@@ -24,9 +24,29 @@ public List<ExtractResult> Extract(string text, DateObject reference)
         {
             var tokens = new List<Token>();
             tokens.AddRange(TimeZoneMatch(text));
+            tokens.AddRange(CityTimeMatch(text));
             return Token.MergeAllTokens(tokens, text, ExtractorName);
         }
 
+        private IEnumerable<Token> CityTimeMatch(string text)
+        {
+            var ret = new List<Token>();
+            var cityMatchResult = config.CityStringMatcher.Find(text.ToLowerInvariant());
```

Comment:
Should try to match suffix first, then list. Otherwise we pay the cost of matching the full list all the time.